exec/java-exec/src/main/codegen/templates/DrillVersionInfo.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+<@pp.dropOutputFile />
+
+<@pp.changeOutputFile name="/org/apache/drill/common/util/DrillVersionInfo.java" />
+
+<#include "/@includes/license.ftl" />
+
+package org.apache.drill.common.util;
+
+import java.text.MessageFormat;
+import java.text.ParseException;
+
+/**
+ * This file is generated with Freemarker using the template src/main/codegen/templates/DrillVersionInfo.java
+ */
+public class DrillVersionInfo {
+  private static final String VERSION = "${maven.project.version}";
+
+  private static final int MAJOR_VERSION = ${maven.project.artifact.selectedVersion.majorVersion};
+  private static final int MINOR_VERSION = ${maven.project.artifact.selectedVersion.minorVersion};
+  private static final int PATCH_VERSION = ${maven.project.artifact.selectedVersion.incrementalVersion};
```

Comment:
yes, that's why I added a big javadoc notice to rely on the version string for semver. Technically we should also add the build version if we wanted to be maven compatible